pkg/controller/machineset/machineset_controller_test.go

```diff
@@ -158,12 +160,18 @@ func TestReconcile(t *testing.T) {
 			}
 
 			defer func() {
-				c.Delete(context.TODO(), tc.instance)
+				err := c.Delete(context.TODO(), tc.instance)
+				if err != nil {
+					t.Logf("error deleting %v: %v", tc.instance, err)
+				}
+
 				select {
 				case recv := <-requests:
 					if recv != tc.expectedRequest {
 						t.Error("received request does not match expected request")
 					}
+
+					t.Logf("Last reconcile %v", <-requests)
```

Comment:
Is this channel guaranteed to receive another request? This seems like an odd thing to add

Reply:
Definitely odd, I couldn't really figure out where is the last request coming from :/. I'd suggest though we focus on following up to replace this completely with ginkgo like coverage (OCPCLOUD-816).